3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -146,7 +147,199 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
+  - *xcoordinate*: numero in virgola mobile che rappresenta la coordinata X di posizione;
+  - *ycoordinate*: numero in virgola mobile che rappresenta la coordinata Y di posizione;
+  - *height*: numero in virgola mobile che rappresenta l'altezza della zona;
+  - *length*: numero in virgola mobile che rappresenta la profondità della zona;
+  - *width*: numero in virgola mobile che rappresenta la larghezza della zona;
+  - *bins*: lista di elementi Bin contenuti nella zona;
+  - *orientation*: booleano che identifica l'orientamento (perpendicolare o parallelo) della zona rispetto all'asse delle ascisse del piano.
+
+  \
+  Per ogni attributo è presente il corrispondente metodo get.
+  Sono inoltre presenti i metodi:
+  - *getLevels*: ritorna una lista contenente le liste di bin che rappresentano i livelli della zona;
+  - *getColumns*: ritorna una lista contenente le liste di bin che rappresentano le colonne della zona;
+  - *getMaxUsedLevel*: ritorna il numero dell'ultimo livello della zona con almeno un bin contenente un prodotto;
+  - *getMaxUsedColumn*: ritorna il numero dell'ultima colonna della zona con almeno un bin contenente un prodotto.
+
+\
+- *Product*:
+
+  rappresenta un prodotto, i suoi attributi sono:
+  - *id*: interno che rappresenta il codice identificativo univoco del prodotto;
+  - *name*: stringa che rappresenta il nome del prodotto;
+  - *weight*: numero in virgola mobile che rappresenta il peso del prodotto;
+  - *height*: numero in virgola mobile che rappresenta l'altezza del prodotto;
+  - *length*: numero in virgola mobile che rappresenta la profondità del prodotto;
+  - *width*: numero in virgola mobile che rappresenta la larghezza del prodotto;
+  - *categories*: lista di stringhe che rappresentano le categorie del prodotto.
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+\
+- *Order*:
+
+  rappresenta la richiesta di uno spostamento di un prodotto tra due bin.
+  I suoi attributi sono:
+  - *id*: interno che rappresenta il codice identificativo univoco della richiesta;
+  - *startPoint*: riferimento al bin iniziale;
+  - *endPoint*: riferimento al bin finale;
+  - *product*: riferimento al prodotto da spostare.
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+\
+- *Floor*:
+
+  rappresenta il piano dell'ambiente 3D, i suoi attributi sono:
+  - *length*: numero in virgola mobile che rappresenta la profondità del piano;
+  - *width*: numero in virgola mobile che rappresenta la larghezza del piano;
+  - *SVG*: stringa che contiene il path al file SVG se presente. Nel caso in cui non sia stato identificato nessun file SVG, la variabile è "null".
+
+  Per ogni attributo è presente il corrispondente metodo get.
+
+  Potendo generare l'oggetto Floor con modalità diverse a seconda della presenza del file SVG, la sua creazione è gestita tramite il design pattern "Strategy" e le relative classi:
+  - FloorStrategyContext;
```

Comment:
Da qui fino alla sezione Three.js compresa, ogni elemento dell'elenco va in monospace e metterei anche una descrizione per ognuno (anche molto breve)